standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/events/ReloadEvent.java

```diff
@@ -67,10 +68,13 @@ public ReloadEvent(String catName, String db, String table, List<String> partVal
                 this.tableObj.getParameters().putAll(tblParams);
             }
             if (partVals != null) {
-                this.ptnObj = handler.get_partition(MetaStoreUtils.prependNotNullCatToDbName(catName, db),
-                        table, partVals);
+                this.ptns = new ArrayList<>();
+                for(List<String> partVal : partVals) {
+                    this.ptns.add(handler.get_partition(MetaStoreUtils.prependNotNullCatToDbName(catName, db),
```

Comment:
how about use `get_partitions_by_names_req` to get all partitions in a round

Reply:
ack